source/_posts/routing-in-react.md

```diff
@@ -0,0 +1,387 @@
+                <Routes> //parent component for Route
+                    <Route exact path="/" element={<Home/>}> //child component
+                        <Route path=":id" element={<Home/>}/>
+                    </Route>
+    
+                    <Route path="/about" element={<About/>}/>
+                    <Route path="/contact" element={<Contact/>}/>
+                </Routes>
+            </BrowserRouter>
+        );
+    }
+    
+    export default App;
+    //import app in index or main file
+
+    ///about.component.jsx
+    //import react first
+    import React from "react";
+    
+    const About = () => {
+        console.log("About");
+        return (
+            <div>
+                <h1>About</h1>
+            </div>
+        );
+    };
+    
+    export default About;
+    //similarly you can create a react component for contact and use export default as follows
+    //export default Contact;
+    
+    ///home.component.jsx
+    import React, {Component, useEffect} from 'react';
+    
+    import {Link} from 'react-router-dom';
+    import {Card, Grid, Paper} from "@mui/material";
+    import {useParams} from "react-router-dom";
+    import axios from "axios";
+    
+    export const userIds = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+    
+    //Home component
+    const Home = ({props}) => {
+        const params = useParams();
+        console.log("params", params);
+        const [userId, setUserId] = React.useState();
+    
+        useEffect(() => {
+            setUserId(params.id);
+        }, [params]);
+    
+    
+        return (
+            <Grid container flexWrap={"nowrap"} spacing={2}>
+                <Grid item xs={3}>
+                    <Paper>
+                        <h2>Users</h2>
+                        {userIds.map(userId => (
+                            <Card key={userId}>
+                                <img src={`https://robohash.org/${userId}`} width={"100"} alt="user"/>
+                                <Link to={`/${userId}`}>User {userId}</Link>
+                            </Card>
+                        ))}
+                    </Paper>
+    
+                </Grid>
+                <Grid item xs={9}>
+                    {userId?<h1>User Id: {userId}</h1>:<h1>Select user</h1>
+                    }
+                </Grid>
+            </Grid>
+        );
+    };
+    
+    export default Home
+
+In this example, we have three routes defined: a home route, an about route, and a contact route. The home route uses a wildcard **path** prop (**/**) to match any path that starts with a **/**, while the about and contact routes use the **exact** prop to specify that the path must be an exact match.
+
+If a user navigates to the **/**path, the home route will be rendered, typically with a home component. If the user navigates to the **/about** path, the about route will be rendered. And if the user navigates to the **/contact** path, the contact route will be rendered.
```

Comment:
A lot of content is set with <strong> here. Please strong only element you really need.